js/scorebord.js

```diff
@@ -1,60 +1,61 @@
-function saveScore() {
-    let name = prompt("Your Name:", "Yohnyy Boy");
- 
-    const userName = name;
-    let userPoints = game.points;
-    
-    let scoreBoard = JSON.parse(localStorage.getItem('results'));
-    
-    if (scoreBoard === null){
-      localStorage.setItem('results', JSON.stringify([{userName, userPoints}]));
-    }	else {
-    	const currentScore = {
-      	userName, 
-        userPoints
-      }
-    	scoreBoard.push(currentScore);
-      localStorage.setItem('results', JSON.stringify(scoreBoard));
-    }
-    
+function saveScore(divName) {
+	const inputTemplate = `
+		<div id="js-scoreIpnut" class="c-input o-flex">
+      <input id="js-input" placeholder="Enter your name" class="c-input__input c-input__item o-flex__item" type="Yout timer name">
+      <div id="js-addBtn" class="c-input__item c-input__button o-flex__item">Add</div>
+		</div>
+		<div id="js-score" class="c-score"></div>
+		`
+	divName.insertAdjacentHTML('beforeend', inputTemplate);
+
 }
-//local storage zapis 
-/*
-function saveScore() {
-	let name = prompt("Your Name:", "Yohnyy Boy");
-	const userName = name;
-	let userPoints = game.points;
-	localStorage.getItem('results')
-	localStorage.setItem("results", JSON.stringify([{userName, userPoints}]));
-};
-	*/
+
 
 function showScore() {
-	
-	let modal = document.querySelector('c-modal');
-	modal.innerHTML = '';
-	let insideModal = document.createElement('div');
-	insideModal.classList.add('c-modal-content' , 'c-score');
-	
-	insideModal.innerText ='Congratulations !!! '+ userName +` You have: ${game.points} points`;
-};
-/*
-// odczyt
-function ReadScores() {
-	localStorage.getItem(game.points);
-	return;
-}
+	const addScore = document.querySelector('#js-addBtn');
+	addScore.addEventListener('click', () => {
+
+		const inputValue = document.querySelector('#js-input');
+		const scoreTable = document.querySelector('#js-score');
 
+		const playerName = inputValue.value;
 
-function scoreBord() {
-	//let userName = document.createElement('div');
+		if (playerName) {
+			const getScore = getItem('results') || [];
+			getScore.push({
+				userName: playerName,
+				userPoints: game.points
+			});
+			const scoreSort = getScore.sort((a, b) => {
+				return b.userPoints - a.userPoints;
+			});
 
-	//userName.innerHTML = name;
-	let modal = document.querySelector('.c-modal');
-	modal.innerHTML = '';
+			if (scoreSort.length > 10) {
+				scoreSort.pop();
+			}
+			scoreSort.forEach((elem, i) => {
```

Comment:
można by użyć destukturyzacji

Reply:
Changed in new commit:
~~~~
scoreSort.forEach(({userName, userPoints}, i) => {});
~~~~